app/controllers/TransferController.scala

```diff
@@ -240,6 +240,7 @@ class TransferController @Inject()(
     implicit request =>
       registrationService.getFinishedData(request.nino) map {
         case NotificationRecord(email) =>
+          cachingService.remove()
```

Comment:
Can you add some unit tests with a verify step to ensure the cache is called only in this scenario and is not called in the others (Mockito verify checks)
We should also consider if we want any logging when removing from the cache fails (recover)